File: 01-week-1--starter-algorithms/04-day-5--find-shortest-string/ruby/find_shortest_string.rb
def find_shortest_string(arr)
  shortest = arr[0]
  arr.each do |value|
    if value.length < shortest.length
      shortest = value
    end
  end
  shortest
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'a'"
  puts "=>", find_shortest_string(['aaa', 'a', 'bb', 'ccc'])

  puts

  puts "Expecting: 'hi'"
  puts "=>", find_shortest_string(['cat', 'hi', 'dog', 'an'])

  puts

  puts "Expecting: 'lily'"
  puts "=>", find_shortest_string(['flower', 'juniper', 'lily', 'dadelion'])

  # Don't forget to add your own!

  # BENCHMARK HERE
end

# Please add your pseudocode to this file

# iterate through each string in the array
# find the length of each string
# store length and string in array of hash
# sort hash by smallest to larget
# return value of first hash in array

#--------------------------------------------
# And a written explanation of your solution

File: 01-week-1--starter-algorithms/04-day-5--find-shortest-string/javascript/find_shortest_string.js
function findShortestString(arr) {
  let shortestArray = arr[0]
  for (let i=0; i<arr.length; i++) {
    if (arr[i].length < shortestArray.length) {
      shortestArray = arr[i]
    }
  }
  return shortestArray
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'a'");
  console.log("=>", findShortestString(['aaa', 'a', 'bb', 'ccc']));

  console.log("");

  console.log("Expecting: 'hi'");
  console.log("=>", findShortestString(['cat', 'hi', 'dog', 'an']));

  console.log("");

  console.log("Expecting: 'lily'");
  console.log("=>", findShortestString(['flower', 'juniper', 'lily', 'dadelion']));

  
  const longArray = [];

  for (let i=0; i<100; i++) {
    longArray.push(Math.random())
  }
  const startTime = Date.now()
  for (let i=0; i<1000; i++) {
    findShortestString(['aa', 'bbb'])
    findShortestString(longArray)
  }

  console.log((Date.now() - startTime) / 2000)
}

module.exports = findShortestString;

// Please add your pseudocode to this file
// And a written explanation of your solution
